src/event-bridge/event-bus.ts

```diff
@@ -103,28 +108,42 @@ export interface IEventBusFilterable<E extends Event> {
    *
    * Unsupported by Functionless:
    * * Variables from outside of the function scope
+   *
+   * @typeParam InEvnt - The type the {@link Rule} matches. Covariant of output {@link OutEvnt}.
+   * @typeParam NewEvnt - The type the predicate narrows to, a sub-type of {@link InEvnt}.
    */
-  when<O extends E>(
+  when<InEvnt extends OutEvnt, NewEvnt extends InEvnt>(
     id: string,
-    predicate: RulePredicateFunction<E, O>
-  ): Rule<E, O>;
-  when<O extends E>(
+    predicate: RulePredicateFunction<InEvnt, NewEvnt>
+  ): Rule<InEvnt, NewEvnt>;
+  when<InEvnt extends OutEvnt, NewEvnt extends InEvnt>(
```

Comment:
These two parameters seem overly complex and redundant. Why did you add them?

Reply:
Need to invert the `in`Evnt` to `out NewEvnt` and `out OutEvnt` to `in InEvnt` for the when method.